### PR TITLE
fix(tokens): limit value of context to defined keys

### LINKED ISSUE #15: Context value is too permissive
__Describe the Bug__

Variable names of Contexts are union of input context + defined keys, when it should only be the defined keys.

__Steps to Reproduce__

Open:

https://nikku.github.io/feel-playground/?e=%7B%0A++foo%3A+15%0A%7D.Mike%27s+age&c=%7E3YCAkIC%2BgICAgICAgIC9AgCXnLILCNLEWVvD4A7RTE6e5VQHnL67RAf8ffGpOTFExGjxp2lIHt%2Fmy8diAzMQSsxIFwmdegBxI7oRrX98i6CA&t=expression

```
{
  foo: 15
}.Mike's age
```
`Mike's age` is parsed as a Variable name on the Context

__Expected Behavior__

`Mike's age` is not a valid variable Name and the expression is parsed as an error

__Environment__

- Host (Browser/Node version), if applicable: [e.g. MS Edge 18, Chrome 69, Node 10 LTS]
- OS: [e.g. Windows 7]
- Library version: [e.g. 2.0.0]

### PR DESCRIPTION
**Pull request by marstamm**
__Which issue does this PR address?__

Closes #15

Note: when we are more restrictive with the context, we now need to define the special name `item` explicitly in filter expressions

### CODE REVIEW

**Comment by nikku** [comment]
I guess we're fixing another issue with this, too: https://nikku.github.io/feel-playground/?e=%5B+1+%5D.Mike%27s+age&c=%7E3YCAkIC%2BgICAgICAgIC9AgCXnLILCNLEWVvD4A7RTE6e5VQHnL67RAf8ffGpOTFExGjxp2lIHt%2Fmy8diAzMQSsxIFwmdegBxI7oRrX98i6CA&t=expression.

Could you test cover that one as well?

File: src/tokens.js [hunk 1]
```diff
@@ -803,13 +797,26 @@ export function trackVariables(context = {}) {
         const children = variables.children.slice(0, -1);
         const lastChild = variables.children.slice(-1)[0];
 
+        let newContext = { ...variables.context, ...lastChild?.computedValue() };
+
+        if (term === pathExpressionStart) {
+          newContext = {
```
Comment (nikku) [inline src/tokens.js:803]: Needs test coverage, cf. https://github.com/nikku/lezer-feel/pull/16#issuecomment-1400362531.

File: src/tokens.js [hunk 2]
```diff
@@ -822,7 +829,19 @@ export function trackVariables(context = {}) {
       }
 
       if (term === ContextEntry) {
-        return wrap(variables, 'ContextEntry', code);
+        const parts = variables.children.filter(c => c.name !== 'ContextEntry');
+
+        const name = parts[0];
+        const value = parts[Math.max(1, parts.length - 1)];
```
Comment (nikku) [inline src/tokens.js:835]: Is there a particular reason this is necessary, i.e. do we assume a context entry to be composed of less than two entries at any point?

Comment (marstamm) [reply src/tokens.js:835]: I thought this was necessary for correct error handling (e.g. `{ foo: }`), but I just confirmed that this also has 2 entries. I'll change it

Comment (nikku) [reply src/tokens.js]: You may want to add a dedicated test case for :arrow_up:, too.

File: src/tokens.js (hunk 1 above)
Comment (marstamm) [reply src/tokens.js:803]: tested in [858b70d](https://github.com/nikku/lezer-feel/pull/16/commits/858b70dfe81cb7035333bea866f12d2458205d23)

File: src/tokens.js (hunk 2 above)
Comment (marstamm) [reply src/tokens.js]: simplified in [6f9f031](https://github.com/nikku/lezer-feel/pull/16/commits/6f9f0316787d66b1093820456e4c4dc66aa97702) and tested in [304ddab](https://github.com/nikku/lezer-feel/pull/16/commits/304ddab9ae9c2b2eb4df7bca15dd8940e5527e13)

File: src/tokens.js (hunk 1 above)
Comment (nikku) [reply src/tokens.js:803]: :star:

**Review by nikku — APPROVED** [review]
Nice job!

**Pull request merged**